assets/app/view/game/part/assignments.rb

```diff
@@ -34,29 +34,84 @@ def preferred_render_locations
 
         def load_from_tile
           @assignments = @tile.hex&.assignments || {}
+          @assignments_to_show = []
+          stack_group_count = {}
+          stack_group_img = {}
+          @assignments.keys.each do |assignment|
+            stack_group = @game.assignment_stack_group(assignment)
+            img = @game.assignment_tokens(assignment, setting_for(:simple_logos, @game))
+
+            stack_group_count[stack_group] = (stack_group_count[stack_group] || 0) + 1 unless stack_group.nil?
+            stack_group_img[stack_group] = img unless stack_group.nil?
+            @assignments_to_show.append({ 'img' => img, 'count' => 1 }) if stack_group.nil?
```

Comment:
Having a single if/else statement is clearer than the repeated unless/if modifiers all testing the same thing. And this is where specifying the default value of zero for the `stack_group_count` hash means that you don't need the slightly awkward test for the existence of the hash key.
```suggestion
            if stack_group
              stack_group_count[stack_group] += 1
              stack_group_img[stack_group] = img
            else
              @assignments_to_show.append({ 'img' => img, 'count' => 1 })
            end
```